### PR TITLE
Feature/247 improve markdown unit

### LINKED ISSUE #247: Improve markdown unit type
Please improve the visualization of markdown input:

image

- [x] Description and Name should be on the same level (same holds true for all the other content types - please fix this for all of 'em)
- [x] The Info/Help-Text needs more padding
- [x] A live preview would be great!
- [x] style markdown in a nother way as normal html (e.g. h1)
- [x] fullscreen edit mode
- [x] (optional) use better texteditor

### PR DESCRIPTION
**Pull request by aeimer**
## Description:
Improves units general form and espacially the markdown unit
Closes #247 

## Improvements
- Better styling
- markdown preview
- markdown html looks different from website-general-html (e.g. a h1)

### CODE REVIEW

File: app/webFrontend/src/app/unit/unit-form/unit-general-info-form/unit-general-info-form.component.html [hunk 1]
```diff
@@ -3,7 +3,11 @@
     <input mdInput required formControlName="name" placeholder="Name">
   </md-input-container>
 
+  <br>
```
Comment (lukaskorte) [inline app/webFrontend/src/app/unit/unit-form/unit-general-info-form/unit-general-info-form.component.html]: Do not use `<br>`

File: app/webFrontend/src/app/unit/unit-form/unit-general-info-form/unit-general-info-form.component.html [hunk 2]
```diff
@@ -3,7 +3,11 @@
...
+
   <md-input-container>
     <textarea mdInput formControlName="description" placeholder="Description"></textarea>
   </md-input-container>
+
+  <hr>
```
Comment (lukaskorte) [inline app/webFrontend/src/app/unit/unit-form/unit-general-info-form/unit-general-info-form.component.html]: Do not use `<hr>`

**Comment by khase** [comment]
In Fullscreen-Mode the TextArea on the Write Tab should also fill the whole visible Space

**Comment by aeimer** [comment]
I know this issue and I have the code that fixes the max-width problem comment out because we are using a  too old version of angular material... Either I do some hacky CSS stuff or we need to wait until we update our dependencies

**Comment by khase** [comment]
It would be nice if the `Write/Preview`-Tab would be always visible in the Fullscreen mode (no matter how far down i've scrolled)

**Comment by aeimer** [comment]
Won't fix that. To much effort...

**Pull request merged**